### PR TITLE
fix(backup): support importing pre-v17 backups (v10-v15 tested, v16 partial)

### PR DESCRIPTION
**Pull request by JehandadK**
## Summary

When v17 rewrote the sync system from PFAPI to operation-log, the cross-model migrations (2 through 4.5) were dropped (`crossModelMigrations: {}`). This means importing any backup created before v17 can fail Typia validation or silently lose data — most critically, **archived tasks are lost** because v17 expects `archiveYoung`/`archiveOld` but older backups have different shapes.

I had a Super Productivity backup from the v10 era (253 tasks, 155 archived tasks, 19 projects). Importing it into v17 silently failed because there was no migration path to bridge the gap.

This PR adds a `migrateLegacyBackup()` function that runs before Typia validation in `BackupService.importCompleteBackup()`, consolidating the essential v16 cross-model migrations into a single pass.

## Version coverage

| Backup source | Detected? | Detection marker | Migrations applied | Status |
|---|---|---|---|---|
| **v10-v12** | Yes | `taskArchive` key | All (2 through 4.5) | **Fully working** |
| **v13** | Yes | `improvement`/`obstruction` keys | 4.2, 4.3, 4.4, 4.5 (2-4.1 skip as idempotent) | **Working** |
| **v14** | Yes | `improvement`/`obstruction` keys | 4.3, 4.4, 4.5 | **Working** |
| **v15** | Yes | `improvement`/`obstruction` keys | 4.4, 4.5 | **Working** |
| **v16** | No | No markers remain | 4.5 (lowercase lang codes only) | **Not detected** — minor gap, only affects uppercase language codes |
| **v17** | No | N/A | None needed | **Native format, unaffected** |

### v13-v15 handling

v13-v15 backups already have `archiveYoung`/`archiveOld` (migration 2 ran in PFAPI), but are detected via `improvement`/`obstruction` keys that were removed in v16. Migration 2's skip condition correctly detects this: it skips when `archiveYoung + archiveOld + timeTracking` all exist without `taskArchive`, preserving existing archived tasks regardless of whether `timeTracking.project` has data.

### Why v16 is not detected

v16 removed the `improvement`/`obstruction` models from its schema, and it never had `taskArchive`. So none of the three detection markers exist in v16 exports. The only missing migration is 4.5 (lowercasing language codes in `localization.lng`), which is cosmetic. A comprehensive fix could read `crossModelVersion` from the backup envelope and run only the needed subset, but that's a larger change.

## What the migrations do

- **Migration 2**: Splits flat `taskArchive` → `archiveYoung`/`archiveOld`, extracts `timeTracking` from project/tag `workStart`/`workEnd`/`breakNr`/`breakTime` properties, normalizes task fields (`issueId` to string, `null` timeEstimate to `0`, `_showSubTasksMode` → `_hideSubTasksMode`)
- **Migration 3**: Migrates planner days → `task.dueDay`, handles legacy INBOX project, removes deprecated `NO_LIST` tag, adds default configs, normalizes falsy task fields
- **Migration 4**: Converts `plannedAt` → `dueWithTime`, strips `TODAY_TAG` from `task.tagIds`
- **Migration 4.1**: Removes `TODAY_TAG` from task repeat config `tagIds`
- **Migration 4.2**: Ensures repeat configs have both `lastTaskCreation` (timestamp) and `lastTaskCreationDay` (date string)
- **Migration 4.3**: Initializes `menuTree` with `projectTree`/`tagTree` arrays
- **Migration 4.4**: Renames `globalConfig.lang` → `globalConfig.localization`, moves `misc.timeLocale` → `localization.dateTimeLocale`, initializes `appFeatures`
- **Migration 4.5**: Lowercases language codes in `localization.lng` and `localization.dateTimeLocale`
- **Final**: Ensures all 18 v17-required keys exist with defaults, strips legacy-only keys

### Detection logic

`isLegacyBackupData()` triggers on three definitive legacy markers:
- `taskArchive` — present in v10-v12 (removed in v13 PFAPI archive split)
- `improvement` — present in v10-v15 (removed in v16)
- `obstruction` — present in v10-v15 (removed in v16)

Modern v17 backups have none of these keys and are completely unaffected.

## Changes

| File | Change |
|------|--------|
| `backup.service.ts` | Import + call legacy migration before `validateFull()` |
| `migrate-legacy-backup.ts` | New — consolidated migration (v16 cross-model 2–4.5) |
| `migrate-legacy-backup.spec.ts` | New — 36 unit tests covering detection, migration, v13-v15 edge case, fixture-based validation, and backward compatibility |
| `test-fixtures/legacy-v10-backup.json` | New — hand-crafted v10-era fixture with edge cases |

## Test plan

- [x] `isLegacyBackupData` correctly detects v10-era backups (has `taskArchive`)
- [x] `isLegacyBackupData` correctly detects backups with `improvement`/`obstruction`
- [x] `isLegacyBackupData` does NOT trigger on modern v17 data
- [x] All 18 v17-required keys present after migration
- [x] All 6 legacy keys stripped after migration
- [x] `taskArchive` tasks migrated into `archiveYoung`
- [x] `archiveOld` initialized empty
- [x] Active task count preserved
- [x] Projects preserved (INBOX added if needed)
- [x] Time tracking extracted from project/tag entities (`workStart` → `s`, `workEnd` → `e`, `breakNr` → `b`, `breakTime` → `bt`)
- [x] `plannedAt` → `dueWithTime` conversion
- [x] Numeric `issueId` → string conversion
- [x] `_showSubTasksMode` → `_hideSubTasksMode` conversion
- [x] `null` timeEstimate/created → `0`
- [x] `menuTree`, `boards`, `issueProvider`, `pluginUserData`, `pluginMetadata` initialized
- [x] `lang` → `localization` config migration + lowercase
- [x] TODAY_TAG removed from repeat config tagIds
- [x] `lastTaskCreationDay` added from `lastTaskCreation`
- [x] Idempotent — re-running on already-migrated data preserves counts
- [x] v13-v15 backups with archiveYoung preserved (empty timeTracking.project edge case)
- [x] v13-v15 backups with empty archiveYoung preserved (no archived tasks)
- [x] Existing `backup.service.spec.ts` tests unaffected
- [x] Full test suite passes: 6,924 tests (Europe/Berlin) + 6,912 tests (America/Los_Angeles)
- [ ] E2E import test with real v10 backup file

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/app/op-log/backup/migrate-legacy-backup.ts [hunk 1]
```diff
@@ -0,0 +1,813 @@
...
+    }
+
+    // Normalize falsy values
+    task.issueId = task.issueId || undefined;
+    task.issueProviderId = task.issueProviderId || undefined;
+    task.issueType =
+      (task.issueType as any) === 'CALENDAR' ? 'ICAL' : task.issueType || undefined;
+    task.issueWasUpdated = task.issueWasUpdated || undefined;
+    task.issueLastUpdated = task.issueLastUpdated || undefined;
+    task.issueAttachmentNr = task.issueAttachmentNr || undefined;
```
Comment (johannesjo) [inline src/app/op-log/backup/migrate-legacy-backup.ts]: ### Code review

Found 1 issue:

1. `||` instead of `??` for numeric fields causes silent data loss when values are `0`. Both `issueAttachmentNr` and `issuePoints` are `number | undefined` fields where `0` is valid (e.g., Jira tasks with 0 attachments or 0 story points). Using `||` converts `0 || undefined` to `undefined`, discarding the data. The fix is to use `??` instead:

```typescript
task.issueAttachmentNr = task.issueAttachmentNr ?? undefined;
task.issuePoints = task.issuePoints ?? undefined;
```

https://github.com/super-productivity/super-productivity/blob/ab9731d055676f4e309a4441e54636f074bdd261/src/app/op-log/backup/migrate-legacy-backup.ts#L517-L521

Note: Lines 524-525 in the same block correctly use `|| 0` for `timeEstimate` and `timeSpent`, which avoids this problem for those fields since the fallback is `0` not `undefined`.

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by johannesjo** [comment]
Thank you very much for this! 

Unfortunately I had to remove the migration code to reduce complexity of the very very complex migration to operation logs. It's great that this PR will bring back legacy compatabiltiy.

**Comment by JehandadK** [comment]
@johannesjo Thanks for the review. This last commit should fix it!

**Comment by johannesjo** [comment]
Thank you very much! <3
